specification/monitor/Microsoft.Monitor.Management/typespec/azureMonitorWorkspace.tsp

```diff
@@ -22,6 +22,11 @@ model AzureMonitorWorkspace
   @path
   @segment("accounts")
   name: string;
+
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-invalid-envelope-property" "ETag is not included in TrackedResource."
+  @doc("Resource entity tag (ETag)")
+  @visibility("read")
+  etag?: string;
```

Comment:
Thanks for your review @razvanbadea-msft.

This PR is just fixing missing attributes from the non-typespec version.

This old property is being in the API since the beginning of the type:
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/monitor/resource-manager/Microsoft.Monitor/stable/2023-04-03/monitoringAccounts_API.json#L501

This API was migrated to typespec for this version some time ago. The missing attribute was found when trying to generate SDKs.

This PR fixes that by adding the missing attribute and fixing the backward compatibility issues for the SDK generation. 